src/Raven.Server/Documents/TimeSeries/StatefulTimeStampValue.cs

```diff
@@ -7,18 +8,87 @@ namespace Raven.Server.Documents.TimeSeries
     public unsafe struct StatefulTimestampValue
     {
         [FieldOffset(0)]
-        public double First;
-        [FieldOffset(8)]
-        public double Last;
+        public long RawFirst;
+        public double First
+        {
+            get
+            {
+                if (RawFirst == 0)
+                    return 0;
+
+                return BitConverter.Int64BitsToDouble(RawFirst);
```

Comment:
Can we apply this only to ARM?

Reply:
Better yet, can we ensure that we are accessing this aligned? That would be better in general

Reply:
I'll run a quick benchmark to see it this entails any performance penalty..

Reply:
Same results.